### PR TITLE
JDK-8300243: Replace NULL with nullptr in share/compiler/

### PR DESCRIPTION
**Pull request by jdksjolen**
Do the conversion in the share/compiler/ sub-directory and all of its files.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300243](https://bugs.openjdk.org/browse/JDK-8300243): Replace NULL with nullptr in share/compiler/

### Reviewers
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**) ⚠️ Review applies to [b23cb6ca](https://git.openjdk.org/jdk/pull/12032/files/b23cb6ca2dd8a871460e9d2973af7274de860ff9)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12032/head:pull/12032` \
`$ git checkout pull/12032`

Update a local copy of the PR: \
`$ git checkout pull/12032` \
`$ git pull https://git.openjdk.org/jdk pull/12032/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12032`

View PR using the GUI difftool: \
`$ git pr show -t 12032`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12032.diff">https://git.openjdk.org/jdk/pull/12032.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/compiler/compiler_globals.hpp [hunk 1]
```diff
@@ -296,10 +296,10 @@
   product(ccstrlist, CompileOnly, "",                                       \
           "List of methods (pkg/class.name) to restrict compilation to")    \
                                                                             \
-  product(ccstr, CompileCommandFile, NULL,                                  \
+  product(ccstr, CompileCommandFile, nullptr,                                  \
```
Comment (vnkozlov) [inline src/hotspot/share/compiler/compiler_globals.hpp:299]: You need to adjust spacing before `\` to align with other lines.

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

**Comment by jdksjolen** [comment]
Thank you for the reviews! I fixed the issues with alignment of `\`  and are now running tests, will integrate when these are done.

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by jdksjolen** [comment]
/integrate

Tests passing, integrating.

**Pull request closed without merging**